File: Lab5/Lab5Sharp/Lab5Sharp/LineWorker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Lab5Sharp
{
    internal static class LineWorker
    {
        public static Line? PerpendicularToAll(Line[] lines)
        {
            for (int indToCheck = 0; indToCheck < lines.Length; indToCheck++)
            {
                Line perpLine = lines[indToCheck];
                bool perpendicular =
                    lines.Where((l, j) => indToCheck != j).All(l => lines[indToCheck].IsPerpendicularTo(l));
                if (perpendicular) return perpLine;
            }

            return null;
        }

        public static void PrintLineArray(Line[] lines, string prePrint = "")
        {
            Console.WriteLine(prePrint);
            Array.ForEach(lines, Console.WriteLine);
            Console.WriteLine();
        }

        public static Line2D GetRandomLine2D(int randMin = -5, int randMax = 6)
        {
            Random random = new Random();
            var a = (random.Next(randMin, randMax), random.Next(randMin, randMax));
            var b = (random.Next(randMin, randMax), random.Next(randMin, randMax));
            return new Line2D(a, b);
        }

        public static Line3D GetRandomLine3D(int randMin = -5, int randMax = 6)
        {
            Random random = new Random();
            var a = (random.Next(randMin, randMax), random.Next(randMin, randMax), random.Next(randMin, randMax));
            var b = (random.Next(randMin, randMax), random.Next(randMin, randMax), random.Next(randMin, randMax));
            return new Line3D(a, b);
        }

        public static Line2D[] GetRandomLine2DArray(int size)
        {
            var lines = new Line2D[size];
            for (int i = 0; i < size; i++)
            {
                lines[i] = GetRandomLine2D();
            }

            return lines;
        }

        public static Line3D[] GetRandomLine3DArray(int size)
        {
            var lines = new Line3D[size];
            for (int i = 0; i < size; i++)
            {
                lines[i] = GetRandomLine3D();
            }

            return lines;
        }
    }
}

File: Lab5/Lab5Python/line_worker.py
from Line import *
from Point import *
from random import randint


def print_list(lst, pre_print=""):
    print(pre_print)
    for item in lst:
        print(item)
    print()


def perpendicular_to_all(lines: list[Line]):
    for ind_to_check in range(len(lines)):
        perp_line = lines[ind_to_check]
        perpendicular = True
        for j in range(len(lines)):
            if ind_to_check != j:
                if not lines[ind_to_check].is_perpendicular_to(lines[j]):
                    perpendicular = False
                    break
        if perpendicular:
            return perp_line


def get_random_line2d(rand_min=-5, rand_max=6) -> Line2D:
    a = Point2D(randint(rand_min, rand_max), randint(rand_min, rand_max))
    b = Point2D(randint(rand_min, rand_max), randint(rand_min, rand_max))
    return Line2D(a, b)


def get_random_line3d(rand_min=-5, rand_max=6) -> Line3D:
    a = Point3D(randint(rand_min, rand_max), randint(rand_min, rand_max), randint(rand_min, rand_max))
    b = Point3D(randint(rand_min, rand_max), randint(rand_min, rand_max), randint(rand_min, rand_max))
    return Line3D(a, b)


def get_random_line2d_list(size) -> list[Line2D]:
    return [get_random_line2d() for i in range(size)]


def get_random_line3d_list(size) -> list[Line3D]:
    return [get_random_line3d() for i in range(size)]
